Common/src/containers/CTrapezoidalMap.cpp

```diff
@@ -0,0 +1,232 @@
+  /* band index */
+  unsigned long i_band = 0;
+  /* number of edges */
+  unsigned long n_edges = edges.size();
+  /* edge index */
+  unsigned long i_edge = 0;
+  /* counter for edges intersects */
+  unsigned long n_intersects = 0;
+  /* lower and upper x value of each band */
+  su2double band_lower_x = 0;
+  su2double band_upper_x = 0;
+
+  su2double x_0;
+  su2double y_0;
+  su2double dy_edge;
+  su2double dx_edge;
+  su2double x_band_mid;
+
+  /* y values of all intersecting edges for every band */
+  y_edge_at_band_mid.resize(unique_bands_x.size() - 1);
+
+  /* loop over bands */
+  while (i_band < n_bands_x) {
+    band_lower_x = unique_bands_x[i_band];
+    band_upper_x = unique_bands_x[i_band + 1];
+    i_edge = 0;
+    n_intersects = 0;
+
+    /* loop over edges and determine which edges appear in current band */
+    while (i_edge < n_edges) {
+      /* check if edge intersects the band
+       * (vertical edges are automatically discarded) */
+      if (((edge_limits_x[i_edge][0] <= band_lower_x) and (edge_limits_x[i_edge][1] >= band_upper_x)) or
+          ((edge_limits_x[i_edge][1] <= band_lower_x) and (edge_limits_x[i_edge][0] >= band_upper_x))) {
+        y_edge_at_band_mid[i_band].push_back(make_pair(0.0, 0));
+
+        x_0 = edge_limits_x[i_edge][0];
+        y_0 = edge_limits_y[i_edge][0];
+
+        dy_edge = edge_limits_y[i_edge][1] - edge_limits_y[i_edge][0];
+        dx_edge = edge_limits_x[i_edge][1] - edge_limits_x[i_edge][0];
+        x_band_mid = (band_lower_x + band_upper_x) / 2.0;
+
+        y_edge_at_band_mid[i_band][n_intersects].first = y_0 + dy_edge / dx_edge * (x_band_mid - x_0);
+
+        /* save edge index so it can later be recalled when searching */
+        y_edge_at_band_mid[i_band][n_intersects].second = i_edge;
+
+        n_intersects++;
+      }
+      i_edge++;
+    }
+
+    /* sort edges by their y values.
+     * note that these y values are unique (i.e. edges cannot
+     * intersect in a band) */
+    sort(y_edge_at_band_mid[i_band].begin(), y_edge_at_band_mid[i_band].end());
+
+    i_band++;
+  }
+
+  su2double duration = ((su2double)clock() - (su2double)build_start) / ((su2double)CLOCKS_PER_SEC);
+
+  if (rank == MASTER_NODE) cout << "Construction of trapezoidal map took " << duration << " seconds\n" << endl;
+}
+
+
+unsigned long CTrapezoidalMap::GetTriangle(su2double val_x, su2double val_y) {
+  /* find x band in which val_x sits */
+  pair<unsigned long, unsigned long> band = GetBand(val_x);
+
+  /* within that band, find edges which enclose the (val_x, val_y) point */
+  pair<unsigned long, unsigned long> edges = GetEdges(band, val_x, val_y);
+
+  /* identify the triangle using the two edges */
+  std::array<unsigned long, 3> triangles_edge_low;
+  
+  for (int i=0;i<2;i++)
+   triangles_edge_low[i] = edge_to_triangle[edges.first][i];
+   
+  std::array<unsigned long, 3> triangles_edge_up;
+  for (int i=0;i<2;i++)
+   triangles_edge_up[i] = edge_to_triangle[edges.second][i];
+
```

Comment:
There is probably a nicer way to initialize an std::array from a vector?
edge_to_triangle is still a vector of vectors. For all mesh points, it stores a vector of the IDs of it's neighboring points.

Reply:
woops, I can't count to 3 apparently...